components/body_sniffer/body_sniffer_url_loader.h

```diff
@@ -124,6 +126,7 @@ class BodySnifferURLLoader : public network::mojom::URLLoaderClient,
   mojo::ScopedDataPipeProducerHandle body_producer_handle_;
   mojo::SimpleWatcher body_consumer_watcher_;
   mojo::SimpleWatcher body_producer_watcher_;
+  mojo::ScopedDataPipeConsumerHandle next_body_consumer_handle_;
```

Comment:
@iefremov @ShivanKaul just pinging you guys here to make sure these changes look ok.

Reply:
So `OnReceiveResponse` and `OnStartLoadingResponseBody` were combined into one, and we need the next_body_consumer_handle_ to keep track of whether it's De-AMP or Speedreader consuming the body?

Reply:
We need to know the consuming handle for the body ahead of time because when the throttles are chained like with De-AMP and Speedreader, if we don't supply the handle then the other throttle(s) in the chain will never receive the body.

Reply:
thanks @mkarolin